### PR TITLE
If we have a JSON body add it to assert message

### PR DESCRIPTION
**Pull request by cjh1**
If assertStatus fails add any JSON response body to the message.

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -196,6 +196,10 @@ def assertStatus(self, response, code):
         """
         code = str(code)
         msg = 'Response status was %s, not %s.' % (response.output_status, code)
+
+        if response.json:
+            msg += ' Response body was %s.' % json.dumps(response.json)
```
Comment (cpatrick) [inline tests/base.py]: @cjh1 I like this. We could go even prettier, though. `json.dumps(response.json, sort_keys=True, indent=4, separators=(',', ': '))`. :dancer:

Comment (cjh1) [inline tests/base.py]: I am happy to make it easier for those that have trouble reading unformatted JSON :smirk_cat: 
Do you think we need a newline before printing the JSON in that case? Something like:
'''
' Response body was:\n%s.' % json.dumps(response.json, sort_keys=True, indent=4, separators=(',', ': '))
'''

Comment (cpatrick) [inline tests/base.py]: Works for me.

**Comment by zachmullen** [comment]
@cjh1 we call `assertStatus` quite a bit, but I'd rather not do all this sorting and pretty printing computation of potentially large objects unconditionally like this. Could we optimize this slightly by only computing this error message after we've tested the status and found it to be invalid? Then instead of doing the `assertTrue` we can just call `self.fail(msg)` inside that condition block.

**Comment by cjh1** [comment]
Sure

**Comment by zachmullen** [comment]
Also, since it looked like you were getting a little squished there: PEP8 allows expressions inside parentheses to be indented on the next line, i.e.

```
msg += ' Response body was:\n%s' % json.dumps(
    response.json, sort_keys=True, indent=4, separators=(',', ': '))
```

**Comment by zachmullen** [comment]
LGTM, thanks. Just verified that this works and looks great:

```
======================================================================
FAIL: testTemporaryPassword (tests.cases.user_test.UserTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/cases/user_test.py", line 477, in testTemporaryPassword
    self.assertStatus(resp, 403)
  File "tests/base.py", line 208, in assertStatus
    self.fail(msg)
AssertionError: Response status was 200 OK, not 403. Response body was:
{
    "message": "Sent temporary access email."
}

----------------------------------------------------------------------
```

**Pull request merged**